### PR TITLE
Cleanup frontend entrypoint

### PR DESCRIPTION
**Pull request by CohenArthur**
This removes a lot of code that was simply ported from the Go frontend, and forbids the usage of multiple input files.

Addresses #1115

### CODE REVIEW

File: gcc/rust/rust-session-manager.cc [hunk 1]
```diff
@@ -574,13 +574,16 @@ Session::enable_dump (std::string arg)
 /* Actual main entry point for front-end. Called from langhook to parse files.
  */
 void
-Session::parse_files (int num_files, const char **files)
+Session::handle_input_files (int num_files, const char **files)
 {
+  if (num_files != 1)
+    rust_fatal_error (Location (),
+		      "only one file may be specified on the command line");
```
Comment (liushuyu) [inline gcc/rust/rust-session-manager.cc:581]: I am not sure if we want to match `rustc`'s error message closely.

`rustc` will emit this message when multiple files are specified:

```
error: multiple input filenames provided (first two filenames are `main.rs` and `lib.rs`)
```

Comment (CohenArthur) [reply gcc/rust/rust-session-manager.cc:581]: I think rustc's message is great :DD We can improve this one of course

**Review by philberty — APPROVED** [review]
LGTM.

I've actually spent alot of time reviewing this PR, the main piece in the compiler driver rustspec.cc we will eventually need to change parts. To handle the no_core and no_std attributes there is got to be a way to send information back here from the compiler proper to say yes link in libcore or libstd.

Maybe during metadata export we export some kind of info into a known file and read info there to figure this out but its not important right now.

**Pull request merged**